app/models/miq_server.rb

```diff
@@ -404,6 +404,74 @@ def logon_status_details
     result.merge(:message => message)
   end
 
+  def ui_hostname
+    if MiqEnvironment::Command.is_podified?
+      ENV.fetch("APPLICATION_DOMAIN")
+    else
+      hostname || ipaddress
+    end
+  end
+
+  def ui_ipaddress
+    if MiqEnvironment::Command.is_podified?
+      nil
+    else
+      ipaddress
+    end
+  end
+
+  def ui_address(contact_with = :hostname)
+    if MiqEnvironment::Command.is_podified?
+      ENV.fetch("APPLICATION_DOMAIN")
+    else
+      contact_with == :hostname ? ui_hostname : ui_ipaddress
+    end
```

Comment:
NOTE it is a bit of duplication to check for is_podified? in both these methods and in ui_address, but with the `contact_with == :hostname` and `::Settings.webservices.contactwith == 'hostname'` checks it seemed cleaner than trying to deal with `contact_with = :ipaddress` but that returning `nil` on podified.

Reply:
The problem is it probably belongs in the kubernetes module for the `MiqServer::WorkerManagement` but we include kubernetes, process, or systemd regardless of the type of server we're running.  I guess this is the only thing you can do.

There are these methods in worker_management though,  although the interface is ugly from here:  `server.worker_manager.class.podified?`

It feels like the worker_manager should know this answer for what type it is and cache the result.  Alas, this code is fine for now.

Reply:
(it would have been nice to just override this method in kubernetes.rb but we include these modules regardless of what type of manager we are)

Reply:
Yeah I didn't think that WorkerManagement was the right place for this since it is more appliance-level but I agree it would be much cleaner to have a k8s subclass for this.  Maybe we do a similar model for EnvironmentManagement (aka convert from simple module to a class then have systemd/kubernetes subclasses)?